policy-format.md

```diff
@@ -0,0 +1,172 @@
+# Explainer: Format of Origin Policy documents
+
+**tl;dr** [Origin Policy](README.md) documents are JSON files with top-level
+entries for each supported policy mechanism.
+
+## General File Format
+
+An origin policy document is a JSON document that contains a series of policy
+items. A top-level dictionary names each policy items and contains item specifc
+calues. Each type of (currently) supported policy item is described below.
+
+Example:
+
+```js
+{
+  "tls": .... ,
+  "content-security-policy": .... ,
+  "referrer": ....
+}
+```
+
+## Format Versioning and Error Handling
+
+Applications should
+
+- raise an error if the file is not well-formed JSON,
+- ignore any top-level section they do not understand,
+- consider it an error if they fail to parse a policy item that they do
+  understand.
+
+**TODO:** Should there be a 'must understand' directive, so that site owners
+can determine themselves whether failure to understand a given policy item
+should be fatal ot nor?
```

Comment:
I think this might be reasonable, but I think it would be a good idea to spell out the cases in which it might be used, and perhaps to point to x.509/TLS as an example. I'm not actually sure what kinds of things are actually marked as critical, and I'm not sure it's worth the complexity, but you're certainly right that if we might want it, we should add it in at the start.

Nit: s/ot nor/or not/ :)

Reply:
(It also presents the question: what do we do when parsing fails? I guess we need to fail the resource load? How do we tell the server that we're not able to support their policy, so that we don't get into an infinite cycle of failing and reloading?)

Reply:
Removed this. I'll answer the questions anyhow, since the matter itself is still unresolved.

Yes. I think the format explainer is the wrong place for *most* of those questions, but it still comes up. The way I currently wrote this:
- General format errors (can't fetch file; not valid JSON) -> fatal error
- top-level item browser doesn't know -> ignore
- format error in item browser does understand -> handle it the same as if the policy was established outside of an origin policy.

This isn't a super workable definition, though, so this will need to be addressed properly, eventually. I hope this is good enough for now.